documentation/proposals/Proposal - Better Strings.md

```diff
@@ -0,0 +1,102 @@
+# Summary
+A rich, more well-defined native string API.
+
+- Written against Silk.NET **1.0.2**.
+- Slated for Silk.NET **2.0.0**.
+
+# Contributors
+- Dylan P, Ultz Limited
+
+# Current Status
+- [x] Proposed
+- [ ] Discussed with API Review Board (ARB)
+- [ ] Approved
+- [ ] Implemented
+
+# Design Decisions
+Currently, the representation of native strings in Silk.NET (and C# as a whole) is an utter mess.
+At the moment we're using `char*` everywhere which isn't technically correct, as a number of users
+have pointed out. This discussion aims to create a richer set of native string APIs to combine
+technical correctness, ease of marshalling, and ease of use.
+
+This is in the form of Char8, Char16, and Char32 which represent the various lengths of characters
+in various encodings. These CharXXs can be combined into a StringXX. All types can be implicitly
+converted to and from their existing C#/.NET types, so that the user doesn't have to deal with much
+
+# Proposed API
+Proposed Silk.NET.Core additions:
+```cs
+public struct Char8
+{
+    private byte _value; // 8-bit character
+    public Char8(byte v) => _value = v;
+    public static implicit operator Char8(char c);
```

Comment:
This isn't necessarily a "safe" implicit conversion.

If this is part of a UTF8 multi-byte character, it would not be correctly converted to `char`.

Reply:
Gotcha, will make it explicit so that overflow exceptions aren't thrown around

Reply:
```suggestion
    public static explicit operator Char8(char c);
```